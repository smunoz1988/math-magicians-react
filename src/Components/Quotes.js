import { useEffect, useState } from 'react';

const Quotes = () => {
  const [loading, setLoading] = useState('not loading');
  const [error, setError] = useState('error');
  const [quotesReceived, setQuote] = useState([]);
  const category = 'funny';

  const fetchData = async () => {
    try {
      const res = await fetch(`https://api.api-ninjas.com/v1/quotes?category=${category}`, {
        headers: {
          'X-Api-Key': '6GbkZeAqM5BjI03ACdq/Ig==y1NOsmUqa5LXYEbO',
        },
      });
      const receiveData = await res.json();
      setQuote(receiveData);
      setError('not error');
    } catch (error) {
      setError('error');
    } finally {
      setLoading('Load');
    }
  };

  useEffect(() => {
    fetchData();
  }, [category]);

  if (loading === 'not loading') {
    return (
      <p className="quoteAuthor">
        Loading
      </p>
    );
  }

  if (error === 'error') {
    return (
      <p className="quoteAuthor">
        Error
      </p>
    );
  }

  return (
    <div role="main" className="quoteContainer">
      {quotesReceived.map((quote) => (
        <div key={quote} className="quotes">
          <p>
            {`"${quote.quote}"`}
          </p>
          <p className="quoteAuthor">
            {quote.author}
          </p>
        </div>
      ))}
    </div>
  );
};

export default Quotes;
