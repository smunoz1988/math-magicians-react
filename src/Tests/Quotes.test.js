import Quotes from '../components/Quotes';
import { render } from '@testing-library/react';

describe('Quotes component', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(<Quotes />);
    expect(asFragment()).toMatchSnapshot();
  });
});
