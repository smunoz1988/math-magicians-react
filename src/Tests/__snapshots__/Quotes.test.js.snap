// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quotes component should match snapshot 1`] = `
<DocumentFragment>
  <p
    class="quoteAuthor"
  >
    Loading
  </p>
</DocumentFragment>
`;
